### DISCUSSION TITLE
Access OIDC token inside tools

### DISCUSSION THREAD

**Discussion by DennisTheD** [open-webui/open-webui#6200] [Q&A]
Hey everybody and thank you for this awesome tool! I have a question related to tool usage. My instances is configured to use an OIDC IdP. This works great. I have some other tools that also utilize this IdP. I am building a custom tool to utilize the HTTP API they provide. I could configure my IdP to add the necessary grants/audiences to the token that gets created when logging into open-webui using SSO. But i currently don't see a way to access users access_token from inside a tool. 

Here some pseudocode fore clarification:
```python
user_valves = __user__.get("valves")
oidc_access_token = __user__.get("idp_access_token") # <-- this is the critical part
endpoint = self.valves.API_ENDPOINT

# use the access_token as authorization for the external service
headers = {"Authorization": "Bearer " + oidc_access_token }
params = {
  # setup some request params here...
}
response = requests.get(endpoint, params=params, headers=headers).json()
``` 

Does the tools API somehow provide access to users access_token? This would make it easy to implement permissions (e.g. for external RAG applications) while keeping all the authentication / authorization related tasks at the IdP.

**Comment by antpyykk-kone** [comment]
This is also something I would like to request. We have been trying to find an easy way in the pipelines framework to access user information. The current solution only provides the very basics ways to identify users. There's also no way to access (afaik) the request made and the authentication information.

The access tokens / refresh tokens are not stored and neither are they available / accessible via cookies.
I also failed to find any real examples of utilizing the user's identity to access resources from external service without having to manually set some user valve (e.g. personal api key).

This prevents us to easily allow pipelines & tools to access user's personal information.
There has been some other related discussion, such as in #5872.

I would like to know if there are plans for this.

**Comment by jlnde** [comment]
Any updates on this feature request? It's crucial to access information of the current users session for a lot of use cases. At least accessing the current username / email within an Open WebUI function would be very beneficial.

 #8957 follows another approach of forwarding the token after SSO.
